fe/fe-core/src/test/java/com/starrocks/sql/plan/PartitionPruneTest.java

```diff
@@ -95,4 +95,26 @@ public void testPredicateEqPrune() throws Exception {
                 "     partitions=1/4\n" +
                 "     rollup: ptest"));
     }
+
+    @Test
+    public void testInClauseCombineOr_1() throws Exception {
+        String plan = getFragmentPlan("select * from ptest where (d2 > '1000-01-01') or (d2 in (null, '2020-01-01'));");
+        assertTrue(plan.contains("  0:OlapScanNode\n" +
+                "     TABLE: ptest\n" +
+                "     PREAGGREGATION: ON\n" +
+                "     PREDICATES: (2: d2 > '1000-01-01') OR (2: d2 IN (NULL, '2020-01-01')), 2: d2 > '1000-01-01'\n" +
+                "     partitions=4/4\n" +
+                "     rollup: ptest"));
+    }
+
+    @Test
+    public void testInClauseCombineOr_2() throws Exception {
+        String plan = getFragmentPlan("select * from ptest where (d2 > '1000-01-01') or (d2 in (null, null));");
+        assertTrue(plan.contains("  0:OlapScanNode\n" +
+                "     TABLE: ptest\n" +
+                "     PREAGGREGATION: ON\n" +
+                "     PREDICATES: (2: d2 > '1000-01-01') OR (2: d2 IN (NULL, NULL)), 2: d2 > '1000-01-01'\n" +
```

Comment:
```suggestion
                "     PREDICATES: (2: d2 > '1000-01-01') OR (2: d2 IN (NULL, NULL)), 2: d2 > '1000-01-01'\n" +
```
We can convert the predicate to NULL when all values are null

Reply:
Yes. My next pr will fix it.